File: src/pnp/compiler/model/instruction/ProcedureInstruction.java
package pnp.compiler.model.instruction;

import pnp.compiler.model.construct.Procedure;
import pnp.compiler.model.construct.type.Type;
import pnp.compiler.model.expression.Expression;

import java.util.List;

public class ProcedureInstruction extends Instruction implements Expression {
    private Procedure procedure;
    private List<Expression> params;

    public ProcedureInstruction(Procedure procedure, List<Expression> params) {
        this.procedure = procedure;
        this.params = params;
    }

    public Procedure getProcedure() {
        return procedure;
    }

    public List<Expression> getParams() {
        return params;
    }

    @Override
    public Type getType() {
        return procedure.getType();
    }
}

File: src/main/kotlin/br/com/pnp/model/instruction/ProcedureInstruction.kt
package br.com.pnp.model.instruction

import br.com.pnp.model.construct.Procedure
import br.com.pnp.model.construct.Variable
import br.com.pnp.model.expression.Expression

class ProcedureInstruction(val procedure: Procedure, val parameters: List<Variable>): Instruction, Expression {
    override val type = procedure.type
}
